pkg/remote/remote.go

```diff
@@ -115,20 +145,40 @@ func NewClient(address string) (Client, error) {
 
 	host := parsedURL.Host
 	pass, _ := parsedURL.User.Password()
-	port := defaultPort
+
+	var port int
 
 	if parsedURL.Port() != "" {
 		// we're safe to ignore error here since `url.Parse` checks if port is valid integer
 		port, _ = strconv.Atoi(parsedURL.Port())
 		host = strings.Split(host, ":")[0]
 	}
 
+	derivedHost := ssh_config.Get(host, "HostName")
+	if derivedHost == "" {
+		derivedHost = host
+	}
+
+	if port == 0 {
+		port, _ = strconv.Atoi(ssh_config.Get(host, "Port"))
+	}
+
+	if port == 0 {
+		port, _ = strconv.Atoi(ssh_config.Get(derivedHost, "Port"))
+	}
+
+	if port == 0 {
+		port, _ = strconv.Atoi(ssh_config.Default("Port"))
```

Comment:
Don't think we need it here (_including str conversion_) since the default value is always port 22. So keeping the initial implementation (_with defaultPort_) seems to be better then this line.